### ISSUE TITLE
Prometheus operator spams logs

### ISSUE THREAD

**Issue by sgmiller** [coreos/prometheus-operator#1274]
**What did you do?**
Followed the getting started guide, as far as getting the Prometheus object created

**What did you expect to see?**

Registration of the prometheus pod.

**What did you see instead? Under which circumstances?**

Rapid spamming of the prometheus-operator log, see below.

**Environment**

* Kubernetes version information:

Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T15:59:24Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T11:55:20Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

  kubeadm

* Manifests:

```
---
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: prometheus-main
spec:
  replicas: 1
  externalUrl: http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/prometheus-main:web/
  serviceMonitorSelector:
    matchLabels:
#      monitored: default
      team: frontend
  resources:
    requests:
      memory: 400Mi
---
apiVersion: apps/v1beta2
kind: Deployment
metadata:
  labels:
    k8s-app: prometheus-operator
  name: prometheus-operator
spec:
  replicas: 1
  selector:
    matchLabels:
      k8s-app: prometheus-operator
  template:
    metadata:
      labels:
        k8s-app: prometheus-operator
    spec:
      containers:
      - args:
        - --kubelet-service=kube-system/kubelet
        - --config-reloader-image=quay.io/coreos/configmap-reload:v0.0.1
        image: quay.io/coreos/prometheus-operator:v0.19.0
        name: prometheus-operator
        ports:
        - containerPort: 8080
          name: http
        resources:
          limits:
            cpu: 200m
            memory: 100Mi
          requests:
            cpu: 100m
            memory: 50Mi
      securityContext:
        runAsNonRoot: true
        runAsUser: 65534
      serviceAccountName: prometheus-operator

```

* Prometheus Operator Logs:

```
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:38.850805468Z caller=operator.go:655 component=prometheusoperator msg="update handler" old=86205090 cur=86205091
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:38.851138282Z caller=operator.go:358 component=alertmanageroperator msg="update handler" old=86205090 cur=86205091
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:38.85174299Z caller=operator.go:690 component=prometheusoperator msg="sync prometheus" key=default/prometheus-main
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:38.925725843Z caller=operator.go:1053 component=prometheusoperator msg="updating config skipped, no configuration change"
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:38.952567737Z caller=operator.go:358 component=alertmanageroperator msg="update handler" old=86205091 cur=86205092
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:38.952715387Z caller=operator.go:655 component=prometheusoperator msg="update handler" old=86205091 cur=86205092
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:38.952759498Z caller=operator.go:690 component=prometheusoperator msg="sync prometheus" key=default/prometheus-main
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.031207968Z caller=operator.go:1053 component=prometheusoperator msg="updating config skipped, no configuration change"
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.056283547Z caller=operator.go:655 component=prometheusoperator msg="update handler" old=86205092 cur=86205093
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.056407295Z caller=operator.go:358 component=alertmanageroperator msg="update handler" old=86205092 cur=86205093
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.056689996Z caller=operator.go:690 component=prometheusoperator msg="sync prometheus" key=default/prometheus-main
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.136198923Z caller=operator.go:1053 component=prometheusoperator msg="updating config skipped, no configuration change"
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.163757289Z caller=operator.go:655 component=prometheusoperator msg="update handler" old=86205093 cur=86205094
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.163785499Z caller=operator.go:358 component=alertmanageroperator msg="update handler" old=86205093 cur=86205094
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.163792029Z caller=operator.go:690 component=prometheusoperator msg="sync prometheus" key=default/prometheus-main
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.322604652Z caller=operator.go:1053 component=prometheusoperator msg="updating config skipped, no configuration change"
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.34890157Z caller=operator.go:655 component=prometheusoperator msg="update handler" old=86205094 cur=86205095
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.349037982Z caller=operator.go:690 component=prometheusoperator msg="sync prometheus" key=default/prometheus-main
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.34925674Z caller=operator.go:358 component=alertmanageroperator msg="update handler" old=86205094 cur=86205095
fprometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.424744117Z caller=operator.go:1053 component=prometheusoperator msg="updating config skipped, no configuration change"
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.450201343Z caller=operator.go:655 component=prometheusoperator msg="update handler" old=86205095 cur=86205097
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.450250053Z caller=operator.go:690 component=prometheusoperator msg="sync prometheus" key=default/prometheus-main
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.450411387Z caller=operator.go:358 component=alertmanageroperator msg="update handler" old=86205095 cur=86205097
gprometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.524406235Z caller=operator.go:1053 component=prometheusoperator msg="updating config skipped, no configuration change"
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.551066445Z caller=operator.go:655 component=prometheusoperator msg="update handler" old=86205097 cur=86205098
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.551205103Z caller=operator.go:358 component=alertmanageroperator msg="update handler" old=86205097 cur=86205098
prometheus-operator-9cf8d896f-nq46t prometheus-operator ts=2018-04-27T15:56:39.551916368Z caller=operator.go:690 component=prometheusoperator msg="sync prometheus" key=default/prometheus-main

```

**Comment by mxinden** [comment]
@sgmiller Would you mind preparing a PR for your suggested changes?

**Comment by brancz** [comment]
Added a PR for leveled logging.

**Comment by brancz** [comment]
#1277 is merged and released (in `v0.20.0`), closing here. Thanks for reporting @sgmiller !
